src/main/java/org/verdictdb/connection/InMemoryAggregate.java

```diff
@@ -0,0 +1,557 @@
+    String sql = selectQueryToSql.toSql(query).toUpperCase();
+    ResultSet rs = conn.createStatement().executeQuery(sql);
+    return new JdbcQueryResult(rs);
+  }
+
+  public static String combine(String combineTableName, String targetTableName, SelectQuery dependentQuery)
+      throws SQLException, VerdictDBException {
+    String tableName = selectAsyncAggTable + selectAsyncAggTableID++;
+
+    // check targetTable exists
+    if (targetTableName.equals("")) {
+      // if not just let it be the copy of combineTable
+      conn.createStatement().execute(
+          String.format("CREATE TABLE %s AS SELECT * FROM %s", tableName, combineTableName));
+    } else {
+      // if exists, combine two tables using the logic of AggCombinerExecutionNode
+      List<GroupingAttribute> groupList = new ArrayList<>();
+      SelectQuery copy = dependentQuery.deepcopy();
+      for (SelectItem sel : copy.getSelectList()) {
+        if (sel instanceof AliasedColumn) {
+          UnnamedColumn col = ((AliasedColumn) sel).getColumn();
+          resetSchemaAndTableForCombine(col);
+          String alias = ((AliasedColumn) sel).getAliasName().toUpperCase();
+          ((AliasedColumn) sel).setAliasName(alias);
+          if (col.isAggregateColumn()) {
+            if (col instanceof ColumnOp && ((ColumnOp) col).getOpType().equals("max")) {
+              ((AliasedColumn) sel).setColumn(new ColumnOp("max", new BaseColumn(alias)));
+            } else if (col instanceof ColumnOp && ((ColumnOp) col).getOpType().equals("min")) {
+              ((AliasedColumn) sel).setColumn(new ColumnOp("min", new BaseColumn(alias)));
+            } else {
+              ((AliasedColumn) sel).setColumn(new ColumnOp("sum", new BaseColumn(alias)));
+            }
+          } else {
+            ((AliasedColumn) sel).setColumn(new BaseColumn(alias));
+            groupList.add(((AliasedColumn) sel).getColumn());
+          }
+        }
+      }
+      SelectQuery left = SelectQuery.create(new AsteriskColumn(),
+          new BaseTable("PUBLIC", targetTableName));
+      SelectQuery right = SelectQuery.create(new AsteriskColumn(),
+          new BaseTable("PUBLIC", combineTableName));
+      AbstractRelation setOperation = new SetOperationRelation(left, right, SetOperationRelation.SetOpType.unionAll);
+      copy.clearFilter();
+      copy.setFromList(Arrays.asList(setOperation));
+      copy.clearGroupby();
+      copy.addGroupby(groupList);
+      String sql = selectQueryToSql.toSql(copy);
+      conn.createStatement().execute(
+          String.format("CREATE TABLE %s AS %s", tableName, sql));
+    }
+
+    return tableName;
+  }
+
+  private static void resetSchemaAndTableForCombine(UnnamedColumn column) {
+    List<UnnamedColumn> columns = new ArrayList<>();
+    columns.add(column);
+    while (!columns.isEmpty()) {
+      UnnamedColumn col = columns.get(0);
+      columns.remove(0);
+      if (col instanceof ColumnOp) {
+
```

Comment:
What is this empty case?

Reply:
I think it should add its operands into list.